src/00/02-introspect/proposal.adoc

```diff
@@ -0,0 +1,879 @@
+of the capabilities it supports:
+
+[source,json]
+----
+include::example-01.json[]
+----
+
+Or could produce this as a JSON output in the case of full version ranges:
+
+[source,json]
+----
+include::example-02.json[]
+----
+
+Which would minimally indicate that the tool only supports the introspection
+capability at versions "1.0.0" through "2.5.0".
+
+Per the findings of the <<User Interface>> research and the consensus of the
+<<SG15: P2717R2 (2023-06-16)>> polling a tool can, additionally, provide the
+introspection information in a file accompanying the tool. There is a challenge
+when providing such an introspection file though: It is not practical to specify
+an absolute location, or locations, across the variety of operating systems and
+tools in the programming ecosystem. As such we provide some possibilities:
+
+1. The name of an introspection file will be the name of the top level invoked
+tool executable (or script, or equivalent) with any type extension
+(i.e. "`.exe`") removed if it exists. That base name will be appended with the
+`.stdinfo` text. For example: `cl.exe` => `cl.stdinfo`, or
+`pass:[g++]` => `pass:[g++].stdinfo`.
+
+2. The introspection file can be found: in the same filesystem location
+as the tool executable (or script, or equivalent), in an implementation defined
+location relative to the tool location, or in an implementation defined global
+location (i.e. an absolute path location).
+
+=== Declaration
+
+The consumer can inform, i.e. declare, to the target tool that specific
+capabilities should use particular versions when responding with information
+using one or more `--std-info=<VersionSpec>` options. The declarations can only
+exist in tandem with options for the mentioned capabilities. It's expected that
+a consumer will first _introspect_ a target tool to discover what it supports.
+Followed by the consumer _declaring_ to the target tool what version(s) of the
+capabilities it is willing to consume. The target tool can then either accept
+the declared capability versions or indicate an error.
+
+An exchange between a consumer and target tool would begin with the
+_introspection_:
+
+[source,shell]
+----
+tool "--std-info"
+----
+
+With a target tool response:
+
+[source,json]
+----
+include::example-03.json[]
+----
+
+Which the consumer can use to _declare_ the specific capability versions:
+
+[source,shell]
+----
+tool "--std-decl=std_info=2.0.0" "--std-decl=gcc_extra=2.1.0" ...
+----
+
+=== Levels
+
+For some use cases it helps to simplify the extent of information the
+introspection understands. While it would be reasonable to expect a tool written
+in a modern general purpose programming language to fully implement all aspects
+of the introspection. It would not be practical to have a shell script parse
+and recognize the more challenging aspect of parsing version number ranges and
+matching them together. To support such use cases the introspection has to
+support levels "min" and "full".
```

Comment:
I would like the bikeshed the use of `min`  here. I think `basic` or `core` are better terms. We don't need to use abbreviations for full words (`min` => `minimum`) otherwise.